lib/index.js

```diff
@@ -984,6 +984,11 @@ Mongoose.prototype.isValidObjectId = function(v) {
   return false;
 };
 
+Mongoose.prototype.syncIndexes = async function() {
+  const result = await this.connection.syncIndexes();
```

Comment:
This won't quite work right in cases where people import mongoose like `const { syncIndexes } = require('mongoose')`

Use the same pattern as in `startSession()`:

```
const _mongoose = this instanceof Mongoose ? this : mongoose;
```